app/state/services.py

```diff
@@ -137,25 +131,26 @@ def get_ip(self, headers: Mapping[str, str]) -> IPAddress:
         return ip
 
 
-def fetch_geoloc_db(ip: IPAddress) -> Geolocation:
-    """Fetch geolocation data based on ip (using local db)."""
-    assert geoloc_db is not None
-
-    res = geoloc_db.city(ip)
-
-    if res.country.iso_code is not None:
-        acronym = res.country.iso_code.lower()
-    else:
-        acronym = "XX"
-
-    return {
-        "latitude": res.location.latitude or 0.0,
-        "longitude": res.location.longitude or 0.0,
-        "country": {
-            "acronym": acronym,
-            "numeric": country_codes[acronym],
-        },
-    }
+def fetch_geoloc_nginx(
+    ip: IPAddress,
+    headers: Mapping[str, str],
+) -> Optional[Geolocation]:
+    """Fetch geolocation data based on ip (using nginx headers)."""
+    country_code = headers.get("X-Country-Code")
+
+    if country_code is not None:
+        country_code = country_code.lower()
+        latitude = headers.get("X-Latitude")
```

Comment:
this will fail if the headers somehow aren't present if we are using `get()` as it will return `None` which cannot be cast to float. unsure if we even want to allow this:

1. either use `["X-Latitude"]`
2. or do `.get("X-Latitude", 0)` or whatever default you wish to use